static/app/components/forms/form.tsx

```diff
@@ -199,6 +199,34 @@ function Form({
     ]
   );
 
+  const getSubmitButtonTitle = useCallback((props: FormModel) => {
+    if (props.isFormIncomplete) {
+      return t('Required fields must be filled out');
+    }
+
+    if (props.isError) {
+      const errors = props.getErrors();
+      const hasRequiredFieldError = [...errors].some(
+        ([_field, message]) => message === fieldIsRequiredMessage
+      );
+
+      if (hasRequiredFieldError) {
+        const allRequiredFieldErrors = [...errors].every(
+          ([_field, message]) => message === fieldIsRequiredMessage
+        );
+
+        if (allRequiredFieldErrors) {
+          return t('Required fields must be filled out');
+        }
+
+        return t('Required fields must be filled out and inputs must be valid');
+      }
+
+      return t('Fields must contain valid inputs');
+    }
+    return undefined;
+  }, []);
```

Comment:
This looks like its trying to be something inbetween a react component and a function that returns the string value title (useCallback seems like a weird choice)

Have you considered taking this fn definition outside of the component or maybe creating a component for this title?

```ts
const SubmitButtonTitle = ({form}: {form: FormModel}) => {...}
title={() => <SubmitButtonTitle form={formModel} />

```

Reply:
It feels weird to create a component just to return a string, so I’ve moved the function outside of `Form` 😉